<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-99.9.9">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Barry Grant">

<title>Class 2 Lab</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="lab2_files/libs/clipboard/clipboard.min.js"></script>
<script src="lab2_files/libs/quarto-html/quarto.js"></script>
<script src="lab2_files/libs/quarto-html/popper.min.js"></script>
<script src="lab2_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="lab2_files/libs/quarto-html/anchor.min.js"></script>
<link href="lab2_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="lab2_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="lab2_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="lab2_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="lab2_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Sections</h2>
   
  <ul>
  <li><a href="#overview" id="toc-overview" class="nav-link active" data-scroll-target="#overview">Overview:</a></li>
  <li><a href="#section-1-dot-plot-parameters" id="toc-section-1-dot-plot-parameters" class="nav-link" data-scroll-target="#section-1-dot-plot-parameters">Section 1: Dot Plot Parameters</a></li>
  <li><a href="#section-2-needleman-wunsch-alignment" id="toc-section-2-needleman-wunsch-alignment" class="nav-link" data-scroll-target="#section-2-needleman-wunsch-alignment">Section 2: Needleman-Wunsch Alignment</a></li>
  <li><a href="#section-3-practice-makes-perfect" id="toc-section-3-practice-makes-perfect" class="nav-link" data-scroll-target="#section-3-practice-makes-perfect">Section 3: Practice makes perfect</a></li>
  <li><a href="#section-4-finding-homologous-sequence" id="toc-section-4-finding-homologous-sequence" class="nav-link" data-scroll-target="#section-4-finding-homologous-sequence">Section 4: Finding homologous sequence</a></li>
  <li><a href="#optional-extension" id="toc-optional-extension" class="nav-link" data-scroll-target="#optional-extension">OPTIONAL EXTENSION</a></li>
  <li><a href="#section-5-the-limits-of-using-blast-for-remote-homologue-detection" id="toc-section-5-the-limits-of-using-blast-for-remote-homologue-detection" class="nav-link" data-scroll-target="#section-5-the-limits-of-using-blast-for-remote-homologue-detection">Section 5: The limits of using BLAST for remote homologue detection</a></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Class 2 Lab</h1>
<p class="subtitle lead">Sequence Alignment &amp; Database Searching (Pt. 1)</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Barry Grant </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<!-- This sets up the PDF form question area style -->
<!-- End style setup -->
<div class="callout-note callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Instructions
</div>
</div>
<div class="callout-body-container callout-body">
<p>Save this document to your computer and open it in a PDF viewer such as Preview (available on every mac) or Adobe Acrobat Reader (<a href="https://get.adobe.com/reader/">free for PC and Linux</a>). Be sure to add your name and UC San Diego personal identification number (PID) and email below before answering all questions in the space provided.<br>
<br>
</p>
</div>
</div>
<section id="overview" class="level2">
<h2 class="anchored" data-anchor-id="overview">Overview:</h2>
<p>Aligning novel sequences with previously characterized genes or proteins provides important insights into their common attributes and evolutionary origins.</p>
<p>In sections <a href="#section-1">1</a>, <a href="#section-2">2</a> and <a href="#section-3">3</a> of this hands-on session we will first explore the principles and methods underlying the computational comparison and alignment of biomolecular sequences.</p>
<p>In section <a href="#section-4">4</a> we explore how these methods are used to search databases to identify homologues sequences (i.e.&nbsp;finding evolutionary related genes or proteins that are descended from a common ancestor).</p>
<p>In section <a href="#section-5">5</a> we highlight the detection limits of conventional BLAST. This sets the scene for introducing more sensitive (but often more time consuming) approaches including Profiles, PSI-BLAST and Hidden Markov Models (HMMs).</p>
</section>
<section id="section-1-dot-plot-parameters" class="level2">
<h2 class="anchored" data-anchor-id="section-1-dot-plot-parameters">Section 1: Dot Plot Parameters</h2>
<p>Dot plots are a simple graphical approach for the visual comparison of two sequences. They have a long history (see <a href="https://pubmed.ncbi.nlm.nih.gov/6801656/">Maizel and Lenk 1981</a> and references therein) and entail placing one sequence on the vertical axis of a 2D grid (or matrix) and the other on the horizontal.</p>
<p>In its simplest form, a dot is placed where the horizontal and vertical sequence values match. More elaborate forms use <em>‘sliding windows’</em> composed of multiple characters and a threshold value, or <em>‘match stringency’</em> for two windows to be considered as matched.</p>
<p>Visit our very own simple dot plot web-app (<a href="https://barryus.shinyapps.io/dotplot/">Link1</a> or it’s mirror <a href="https://bioboot.shinyapps.io/dotplot/">Link2</a>) and get a feel for how altering these major dot plot parameters change the displayed protein and DNA dot plots.</p>
<blockquote class="blockquote">
<p>N.B. Note the questions listed on the web page (also found below) and add your answers in the space provided on the next page.</p>
</blockquote>
<p><br>
</p>
<div id="fig-dotplot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<figure class="figure">
<img src="img/dot_pot_app.png" title="Title: Interactive Dot Plot Web App" class="img-fluid figure-img">
</figure>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: Interactive dot plot web app to explore fundemental concepts of biomolecular sequence comparasion</figcaption><p></p>
</figure>
</div>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q1</strong> Why does the DNA sequence have more dots than the protein sequence plot?</p>
<div class="callout-tip callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Tip
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>What do you know about DNA composition vs protein composition?</p>
</div>
</div>
</div>
</div>
</div>
</div>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q2</strong> How can we increase the signal to noise ratio?</p>
<div class="callout-tip callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Tip
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Signal in this case means correct matches that we actually want to highlight and noise means spurious matches that we don’t want.</p>
</div>
</div>
</div>
</div>
</div>
</div>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q3</strong> What does a ‘Match stringency’ larger than ‘Window size’ yield and why?</p>
</div>
</div>
</div>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q4</strong> What are the major weaknesses of this approach?</p>
<div class="callout-tip callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Tip
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Is your inner nerd happy with this approach? How would you use it to determine if a second set of sequences was more similar to each other than a first set of sequences?</p>
</div>
</div>
</div>
</div>
</div>
</div>
<p><br>
</p>
</section>
<section id="section-2-needleman-wunsch-alignment" class="level2">
<h2 class="anchored" data-anchor-id="section-2-needleman-wunsch-alignment">Section 2: Needleman-Wunsch Alignment</h2>
<p>Sequence alignment methods often use something called a ‘dynamic programming’ algorithm that can be usefully considered as an extension of the dot plot approach. Here we have two sample sequences, and we’d like to use the Needleman-Wunsch algorithm discussed in class to align them. Feel free to use the clasroom white-boards and/or pen and paper and attach a photo to this PDF for gradescope.</p>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q5</strong> Using a <strong>match score</strong> of <strong>+2</strong>, a <strong>mismatch score</strong> of <strong>-1</strong>, and a <strong>gap score</strong> of <strong>-2</strong>. Fill in the table below (or use pen and paper) for the following two sequences:</p>
<p><br>
</p>
<p>Sequence 1: ATTGC<br>
Sequence 2: AGTTC</p>
<p><br>
</p>
<p>From your completed table what is the <strong>optimal score</strong> and corresponding <strong>alignment</strong> (with one sequence above the other)?</p>
<div class="callout-tip callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Tip
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>It can be hard to store the all important progress arrows in the PDF version of this document and thus you may prefer to use your own paper (or white-board) version that you can take a photo off for upload to gradescope.</p>
</div>
</div>
</div>
<p><br>
</p>
</div>
</div>
</div>
<p><br>
</p>
</section>
<section id="section-3-practice-makes-perfect" class="level2">
<h2 class="anchored" data-anchor-id="section-3-practice-makes-perfect">Section 3: Practice makes perfect</h2>
<p>Again use the Needleman-Wunsch algorithm discussed in class to align the following sequences:</p>
<p>&nbsp;</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q6</strong> Using a <strong>match score</strong> of <strong>+2</strong>, a <strong>mismatch score</strong> of <strong>-1</strong>, and a <strong>gap score</strong> of <strong>-2</strong>. Fill in the table below and translate it into a alignment. What is the optimal score for this alignment? <strong>Is there one unique alignment with this score?</strong></p>
<p><br>
</p>
<p>Sequence 1: TATAG<br>
Sequence 2: GTTAC</p>
<p><br>
</p>
</div>
</div>
</div>
<p><br>
</p>
<p><br>
</p>
</section>
<section id="section-4-finding-homologous-sequence" class="level2">
<h2 class="anchored" data-anchor-id="section-4-finding-homologous-sequence">Section 4: Finding homologous sequence</h2>
<p>Your collaborators found a protein while working on a fly species and have asked you to see if there are any human homologs.</p>
<pre><code>&gt;fly_protein
MDNHSSVPWASAASVTCLSLDAKCHSSSSSSSSKSAASSISAIPQEETQTMRHIAHTQRCLSRLTSLVAL
LLIVLPMVFSPAHSCGPGRGLGRHRARNLYPLVLKQTIPNLSEYTNSASGPLEGVIRRDSPKFKDLVPNY
NRDILFRDEEGTGADRLMSKRCKEKLNVLAYSVMNEWPGIRLLVTESWDEDYHHGQESLHYEGRAVTIAT
SDRDQSKYGMLARLAVEAGFDWVSYVSRRHIYCSVKSDSSISSHVHGCFTPESTALLESGVRKPLGELSI
GDRVLSMTANGQAVYSEVILFMDRNLEQMQNFVQLHTDGGAVLTVTPAHLVSVWQPESQKLTFVFADRIE
EKNQVLVRDVETGELRPQRVVKVGSVRSKGVVAPLTREGTIVVNSVAASCYAVINSQSLAHWGLAPMRLL
STLEAWLPAKEQLHSSPKVVSSAQQQNGIHWYANALYKVKDYVLPQSWRHD</code></pre>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q7</strong> Using the default settings for NCBI BLAST, can you find any homologs for this protein in Humans?</p>
<div class="callout-tip callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Tip
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Try using the LIMITS and FILTERING options we covered in the last lab.</p>
</div>
</div>
</div>
</div>
</div>
</div>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q8</strong> Try changing the database to <em>refseq_protein</em>. From the results, select a few proteins and find the common name for the species. What trend do you notice as you move down the results list?</p>
<div class="callout-tip callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Tip
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Search google for the species name and use the taxonomy tab on your NCBI BLAST results page.</p>
</div>
</div>
</div>
</div>
</div>
</div>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q9</strong> Finally, try also limiting the search to only <em>H. Sapiens</em>. What function do these proteins have?</p>
<div class="callout-tip callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Tip
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>You can simply type the <em>Taxon ID</em> <strong>9606</strong> in the “Organism” box.</p>
</div>
</div>
</div>
</div>
</div>
</div>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q10</strong> What function do you think this protein performs for your collaborators’ organism?</p>
</div>
</div>
</div>
<p><br>
</p>
</section>
<section id="optional-extension" class="level2">
<h2 class="anchored" data-anchor-id="optional-extension">OPTIONAL EXTENSION</h2>
<p>We will revisit this problem and introduce approaches with greater sensitivity (i.e.&nbsp;ability to find more remote homolouges) in the next lab.</p>
</section>
<section id="section-5-the-limits-of-using-blast-for-remote-homologue-detection" class="level2">
<h2 class="anchored" data-anchor-id="section-5-the-limits-of-using-blast-for-remote-homologue-detection">Section 5: The limits of using BLAST for remote homologue detection</h2>
<p>Let’s return to the HBB protein that we explored in a previous class and see if we can find distantly related myoglobin and neuroglobin using HBB as a BLAST query.</p>
<pre><code>&gt;gi|4504349|ref|NP_000509.1| hemoglobin subunit beta [Homo sapiens]
MVHLTPEEKSAVTALWGKVNVDEVGGEALGRLLVVYPWTQRFFESFGDLSTPDAVMGNPKVKAHGKKVLG
AFSDGLAHLDNLKGTFATLSELHCDKLHVDPENFRLLGNVLVCVLAHHFGKEFTPPVQAAYQKVVAGVAN
ALAHKYH</code></pre>
<p>After selecting <strong>blastp</strong> and entering the sequence, be sure to change the search database to “<strong>refseq-protein</strong>” and restrict our search organism to only <strong>humans</strong> (taxid: 9605). This will help focus our results to highlight distant homologs in humans.</p>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q11</strong> What homologs did you find with this simple blastp search? Note their percent identities, coverage and E-values.</p>
</div>
</div>
</div>
<p><br>
</p>
<p>Now we could try changing the <strong>Algorithm parameters</strong> on the submission page to increase the number of hits reported. To do this you can click on the <strong>Edit and Resubmit</strong> link at the top left of your results page.</p>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q12</strong> Try increasing the Expect threshold for your blasts search (e.g.&nbsp;to 2000). What new hits were reported? What about their alignment statistics? Do you trust these matches? Did you find myoglobin?</p>
</div>
</div>
</div>
<p><br>
</p>
<p><br>
</p>
<div class="callout-important callout callout-style-simple no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p><strong>Q13</strong>: What one part of this exercise or associated lecture material is still confusing?</p>
<p>If appropriate please also indicate the question number from this document and answer the question in the following anonymous form: <a href="https://forms.gle/XFgux3J6pU34Qeen6"><strong>Mudy_Point_Assesment_Form</strong></a></p>
<p>Your comments will let us know which material needs to be further clarified and will help us gain stronger control of the material in this course. Thank you!</p>
</div>
</div>
</div>
<p><br>
</p>
</section>
<section id="discussion" class="level2">
<h2 class="anchored" data-anchor-id="discussion">Discussion</h2>
<p>Many useful ‘rules of thumb’ are expressed in terms of precent identity. If two proteins have more than 45% identical residues in their optimal alignment they typically have very similar structures and are likely to have a similar function. If two proteins have more than 25% identical residues (but less than 45% identity), they are likely to have a similar general folding pattern. Note that we will expand on the basis of this important <em>sequence &gt; structure &gt; function</em> relationship in the next lab.</p>
<p>Observations of a lower degree of sequence similarity cannot however rule out homology. Our very own late <a href="https://en.wikipedia.org/wiki/Russell_Doolittle">Russ Doolittle</a> defined the region between 18-25% sequence identity as the “twilight zone” in which the suggestion of homology is tantalizing but dangerous. Below the <strong>twilight zone</strong> is a region where pairwise sequence alignments tell us very little - sometimes called the “midnight zone”.</p>
<p>Our next class will introduce more advanced topics including profile and structure based approaches that can delve deeper into these important, but often hard to detect, sequence-structure-function relationships.</p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>